scripts/filter_expected_warnings.py

```diff
@@ -55,10 +55,11 @@ def __init__(self, all_lines):
             elif line.startswith('  '):
                 current.otherlines.append(line.strip())
             else:
-                # Assuming all warnings are of the form [path:line: warning:...]
+                # Warnings are usually of the form [path:line: warning:...]
                 # (and the warnings about too many nodes have been filtered out).
-                print('Error filtering warnings: Unexpected input format.')
-                print('  Input:' + line)
+                # Treat any unexpected input lines as warnings.
+                current = DoxygenWarning(line, '', line)
```

Comment:
:pick: Probably add another constructor to `DoxygenWarning` to allow creating from an unrecognised format?

Reply:
I tried (briefly) to do that, but it didn't work on my first attempts.  (I found this link https://stackoverflow.com/questions/682504/what-is-a-clean-pythonic-way-to-have-multiple-constructors-in-python )